lib/infrastructure/repositories/messenger_repository.dart

```diff
@@ -155,51 +155,38 @@ class MessengerRepository
           discussionName: discussionName,
         );
 
-        if (membersAddedToNotify != null && membersAddedToNotify.isNotEmpty) {
-          await _sendTransactionNotification(
-            notificationRecipientAddress: updatedDiscussion.discussion.address,
-            listenAddresses: membersPubKeys,
-            creator: owner,
-            session: session,
-            previousKeyPair: updatedDiscussion.previousKeyPair,
-            pushNotification: {
-              'en': const PushNotification(
-                title: 'Archethic',
-                body: 'You have been added to a discussion',
-              ),
-              'fr': const PushNotification(
-                title: 'Archethic',
-                body: 'Vous avez été ajouté à une discussion',
-              ),
-            },
-            transactionType:
-                MessengerConstants.notificationTypeNewDiscussionAdding,
-          );
-        }
-
-        if (membersDeletedToNotify != null &&
+        var listenAddresses = membersPubKeys;
+        // If there are users added or deleted, we will notify only them. Otherwise (like a name changed), we will notify everybody.
+        if (membersAddedToNotify.isNotEmpty ||
             membersDeletedToNotify.isNotEmpty) {
-          await _sendTransactionNotification(
-            notificationRecipientAddress: updatedDiscussion.discussion.address,
-            listenAddresses: membersPubKeys,
-            creator: owner,
-            session: session,
-            previousKeyPair: updatedDiscussion.previousKeyPair,
-            pushNotification: {
-              'en': const PushNotification(
-                title: 'Archethic',
-                body: 'You have been removed from a discussion',
-              ),
-              'fr': const PushNotification(
-                title: 'Archethic',
-                body: "Vous avez été supprimé d'une discussion",
-              ),
-            },
-            transactionType:
-                MessengerConstants.notificationTypeNewDiscussionDeletion,
-          );
+          // https://stackoverflow.com/questions/21826342/how-do-i-combine-two-lists-in-dart with If you want to merge two lists and remove duplicates
```

Comment:
add all your elements with doublon in a list and do that to remove doublons:
 xxxxx.toSet().toList(),